### PR TITLE
Lf 4157 create add more animals card

### PR DESCRIPTION
**Pull request by Duncan-Brain**
**Description**

NOTE: Branched off of buggy icons branch for icon reuse -- but the diff sucks because its behind integration 😂 .

I think this is mostly done -- I was noticing storybook was not overriding styles correctly -- I would be interested if you experience the same thing because it looks fine if added in any random app page.

Two questions I haven't resolved:
- I asked Loic about reusing one of the new buttons styles instead of as designed.
- @SayakaOno it is scoped to accept a way to change width/height of card. I was not seeing my styles override as expected when passed in through a className so I made a generic card to wrap this one for now. Thought maybe it could be used by both cards if that made sense?  

Jira link: [LF-4157](https://lite-farm.atlassian.net/browse/LF-4157)

**Type of change**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Passes test case
- [ ] UI components visually reviewed on desktop view
- [ ] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [ ] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

[LF-4157]: https://lite-farm.atlassian.net/browse/LF-4157?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/webapp/src/components/Animals/AddAnimalsForm/styles.module.scss [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  background: var(--White, #FFF);
+}
+
+.animalIcons {
+  @include svgColorFill(var(--Colors-Primary-Primary-teal-600))
+}
+
+.addMoreText {
+  width: 304px;
+  color: var(--Colors-Neutral-Neutral-900, #2B303A);
```
Comment (kathyavini) [inline packages/webapp/src/components/Animals/AddAnimalsForm/styles.module.scss:40]:
```suggestion
  border: 1px solid var(--Colors-Primary-Primary-teal-50);
  background: var(--White);
}

.animalIcons {
  @include svgColorFill(var(--Colors-Primary-Primary-teal-600))
}

.addMoreText {
  width: 304px;
  color: var(--Colors-Neutral-Neutral-900);
```

File: packages/webapp/src/components/Animals/AddAnimalsForm/styles.module.scss [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  font-style: normal;
+  font-weight: 400;
+  line-height: normal;
+}
+
+.addMoreButton {
+  text-decoration-color: var(--Colors-Primary-Primary-teal-600);
+}
+
+.buttonIcon, .buttonIcon:hover {
```
Comment (kathyavini) [inline packages/webapp/src/components/Animals/AddAnimalsForm/styles.module.scss:53]:
```suggestion
.buttonIcon, .buttonIcon:hover, .addMoreCard:active .buttonIcon {
```
To fix this when clicking:

Screenshot 2024-04-11 at 1 55 51 PM

File: packages/webapp/public/locales/en/translation.json [hunk 3]
```diff
@@ -156,6 +156,8 @@
     "WILD_CROP": "Wild crops"
   },
   "ANIMAL": {
+    "ADD_MORE_BODY_TEXT": "Do you own other animal type or a different breed?",
```
Comment (kathyavini) [inline packages/webapp/public/locales/en/translation.json:159]:
```suggestion
    "ADD_MORE_BODY_TEXT": "Do you own another animal type or a different breed?",
```

Oh dear, there are two different texts for this depending on where in Figma you look 😂 Also neither is perfectly grammatical. I'll ping Loïc to see if he wants "own" or "have" here.

File: packages/webapp/src/components/Animals/AddAnimalsForm/MoreAnimalsCard.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import AddAnimalsCard from './AddAnimalsCard';
+import styles from './styles.module.scss';
+import { useTranslation } from 'react-i18next';
+import clsx, { ClassValue } from 'clsx';
```
Comment (kathyavini) [inline packages/webapp/src/components/Animals/AddAnimalsForm/MoreAnimalsCard.tsx:19]:
```suggestion
import { ClassValue } from 'clsx';
```

The `className` prop isn't actually used by the component. Not sure if it's an oversight and we should build it in (apply it somewhere), or delete it entirely... probably delete, right? 😅

**Review by kathyavini — COMMENTED** [review]
> I was noticing storybook was not overriding styles correctly -- I would be interested if you experience the same thing because it looks fine if added in any random app page.

I have seen this too for a few components. I'm not sure which styles are problematic here though? @SayakaOno do you know? It looks like the Figma to me, even in Storybook 😂 

> @SayakaOno it is scoped to accept a way to change width/height of card. I was not seeing my styles override as expected when passed in through a className so I made a generic card to wrap this one for now. Thought maybe it could be used by both cards if that made sense?

I thought the outcome of our short tech daily discussion on this ticket (back before Anto left) was that we're not passing props or anything, just letting the card fill the parent container 🤔 I think the implementation with the dummy wrapper component (which is going to have to go at some point I guess?) is consistent with what we had discussed.

Edit: But personally I would prefer the wrapper and its styles to exist in the Storybook story, not as a component in App that we later delete!!

File: packages/webapp/src/components/Animals/AddAnimalsForm/MoreAnimalsCard.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
+/*
+ *  Copyright 2023 LiteFarm.org
```
Comment (kathyavini) [inline packages/webapp/src/components/Animals/AddAnimalsForm/MoreAnimalsCard.tsx:2]:
```suggestion
 *  Copyright 2024 LiteFarm.org
```

File: packages/webapp/src/components/Animals/AddAnimalsForm/styles.module.scss [hunk 6]
```diff
@@ -0,0 +1,55 @@
+/*
+ *  Copyright 2023 LiteFarm.org
```
Comment (kathyavini) [inline packages/webapp/src/components/Animals/AddAnimalsForm/styles.module.scss:2]:
```suggestion
 *  Copyright 2024 LiteFarm.org
```

**Review by kathyavini — COMMENTED** [review]
Missed some license years.

File: packages/webapp/src/components/Icons/styles.module.scss [hunk 7]
```diff
@@ -34,4 +35,23 @@
...
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  gap: 8px;
+}
+
+.pill {
+  border-radius: 64px;
+  -moz-border-radius: 64px;
+  -webkit-border-radius: 64px;
```
Comment (SayakaOno) [inline packages/webapp/src/components/Icons/styles.module.scss:56]: Do we need these??

File: packages/webapp/src/components/Animals/AddAnimalsForm/MoreAnimalsCard.tsx [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+import AddAnimalsCard from './AddAnimalsCard';
+import styles from './styles.module.scss';
+import { useTranslation } from 'react-i18next';
+import { ClassValue } from 'clsx';
+import Icon, { Icons } from '../../Icons';
+import Button from '../../Form/Button';
+
+type MoreAnimalCardProps = {
+  className: ClassValue;
```
Comment (SayakaOno) [inline packages/webapp/src/components/Animals/AddAnimalsForm/MoreAnimalsCard.tsx:24]:
```suggestion
  className?: ClassValue;
```

**Review by SayakaOno — COMMENTED** [review]
It seems like a storybook bug to me that styles are not applied properly. I'll leave a note here so we can keep this in mind.

```
<div className={clsx(styles.icons, pill && styles.pill, className)}>{CustomIcons}</div>
```
`styles.icons` is supposed to be overriden by `className`.

In the app:
Screenshot 2024-04-12 at 9 38 07 AM

Storybook:
Screenshot 2024-04-12 at 9 38 25 AM

> I thought the outcome of our short tech daily discussion on this ticket (back before Anto left) was that we're not passing props or anything, just letting the card fill the parent container 🤔

@kathyavini I was picturing the same, and I agree with what you mentioned:

> I propose deleting this file entirely though and moving the card styling (minus width and height) to MoreAnimalsCard. I don't think Card component is necessary. Width and height could be set using a wrapper in the Storybook story.

There's a few things I would like to update, so I will create a PR against this PR!

File: packages/webapp/src/components/Icons/icons.jsx [hunk 9]
```diff
@@ -98,6 +102,9 @@ const iconMap = {
   REMOVE_ANIMAL: RemoveAnimalIcon,
   // Tasks
   SOIL_AMENDMENT_TASK: SoilAmendmentTask,
+  // System
```
[the remark this answers is not in the document]
Comment (kathyavini) [reply packages/webapp/src/components/Icons/icons.jsx:105]: Sounds good. I wasn't sure if it was only icons that were problematic in some way that needed the component (that would be something to ask Duncan when he returns!) But I'm fine to keep going with all new icons.

File: packages/webapp/src/components/Icons/styles.module.scss (hunk 7 above)
Comment (kathyavini) [reply packages/webapp/src/components/Icons/styles.module.scss:56]: No!

File: packages/webapp/src/components/Icons/icons.jsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (SayakaOno) [reply packages/webapp/src/components/Icons/icons.jsx:105]: Oh I see! Sounds good!

**Review by SayakaOno — APPROVED** [review]

**Review by kathyavini — APPROVED** [review]

**Pull request merged**